src/Microsoft.PowerShell.Commands.Utility/commands/utility/ReadConsoleCmdlet.cs

```diff
@@ -144,11 +155,26 @@ protected override void BeginProcessing()
             }
             else
             {
+                if (this.MyInvocation.BoundParameters.ContainsKey(nameof(AsSecureString)) && this.MyInvocation.BoundParameters.ContainsKey(nameof(MaskInput)))
+                {
+                    InvalidOperationException exception = new InvalidOperationException(ReadHostStrings.CannotSpecifyAsSecureStringAndMaskInput);
+                    var errorRecord = new ErrorRecord(
+                        exception,
+                        "CannotSpecifyAsSecureStringAndMaskInput",
+                        ErrorCategory.InvalidOperation,
+                        targetObject: null);
+
+                    ThrowTerminatingError(errorRecord);
+                }
```

Comment:
Wouldn't this be better implemented by having `-MaskInput` and `-AsSecureString` in separate parameter sets?

That way the error would be thrown from the parameter binder if you try to use both, rather than waiting for the cmdlet to finish initialization first.

Reply:
Thanks for the comment!

My motivation was that the error that I could generate when validating parameters at runtime is more helpful than the one that gets generated from parameters sets. IIRC there are some issues on that, so it will be improved soon.

I changed the implementation in 7f21ad7e82a2ff5f2e632ba984285c873eb0331b